src/highlight/HighlightAnnotations.tsx

```diff
@@ -1,45 +1,170 @@
 import * as React from 'react';
-import HighlightCanvas from './HighlightCanvas';
+import noop from 'lodash/noop';
 import HighlightCreator from './HighlightCreator';
 import HighlightList from './HighlightList';
-import { AnnotationHighlight } from '../@types';
-import { isValidHighlight, sortHighlight } from './highlightUtil';
-
+import PopupReply from '../components/Popups/PopupReply';
+import SingleHighlightAnnotation from './SingleHighlightAnnotation';
+import { AnnotationHighlight, Rect } from '../@types';
+import { CreatorHighlight, CreatorStatus, Mode } from '../store';
 import './HighlightAnnotations.scss';
 
+type CreateArg = {
+    location: number;
+    message: string;
+    shapes: Rect[];
+    text?: string;
+};
+
 type Props = {
     activeAnnotationId: string | null;
     annotations: AnnotationHighlight[];
+    createHighlight?: (arg: CreateArg) => void;
     isCreating: boolean;
+    location: number;
+    message: string;
     setActiveAnnotationId: (annotationId: string | null) => void;
+    setMessage: (message: string) => void;
+    setMode: (mode: Mode) => void;
+    setStaged: (staged: CreatorHighlight | null) => void;
+    setStatus: (status: CreatorStatus) => void;
+    staged?: CreatorHighlight | null;
+    status: CreatorStatus;
 };
 
-export default class HighlightAnnotations extends React.PureComponent<Props> {
+type State = {
+    highlightRef?: HTMLAnchorElement;
+};
+
+export function getStagedAnnotation(
```

Comment:
This looks like it should be a selector, but can we store the staged annotation in this format so we don't need to transform it each time it's rendered?